### PR TITLE
chore(server): save original file name with extension

### PR DESCRIPTION
**Pull request by alextran1502**

### CODE REVIEW

File: server/src/immich/api-v1/asset/asset.service.ts [hunk 1]
```diff
@@ -356,7 +356,7 @@ export class AssetService {
       duration: dto.duration || null,
       isVisible: dto.isVisible ?? true,
       livePhotoVideo: livePhotoAssetId === null ? null : ({ id: livePhotoAssetId } as AssetEntity),
-      originalFileName: parse(file.originalName).name,
+      originalFileName: parse(file.originalName).name + parse(file.originalName).ext,
```
Comment (jrasm91) [inline server/src/immich/api-v1/asset/asset.service.ts:359]: We should pass this through unmodified now.

**Review by jrasm91 — COMMENTED** [review]
Is this still WIP? I don't see changes to the web or mobile to not append the extension anymore. I also don't see a migration.

File: server/src/infra/migrations/1709763765506-AddExtensionToOriginalFileName.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddExtensionToOriginalFileName1709763765506 implements MigrationInterface {
```
Comment (alextran1502) [inline server/src/infra/migrations/1709763765506-AddExtensionToOriginalFileName.ts:3]: @jrasm91 Migration is added

@mertalev Does the query look ok here to append the file extension to the original name?

File: web/src/lib/utils/asset-utils.ts [hunk 3]
```diff
@@ -102,14 +102,14 @@ export const downloadFile = async (asset: AssetResponseDto) => {
   }
   const assets = [
     {
-      filename: `${asset.originalFileName}.${getFilenameExtension(asset.originalPath)}`,
+      filename: `${asset.originalFileName}`,
```
Comment (mertalev) [inline web/src/lib/utils/asset-utils.ts:105]: You can just pass `asset.originalFileName` now without the `${}` stuff (below as well)

Comment (alextran1502) [reply web/src/lib/utils/asset-utils.ts:105]: good catch

File: server/src/domain/download/download.service.ts [hunk 4]
```diff
@@ -91,12 +91,12 @@ export class DownloadService {
...
       const { originalPath, originalFileName } = asset;
-      const extension = extname(originalPath);
-      let filename = `${originalFileName}${extension}`;
+
+      let filename = originalFileName;
       const count = paths[filename] || 0;
       paths[filename] = count + 1;
       if (count !== 0) {
-        filename = `${originalFileName}+${count}${extension}`;
+        filename = `${parse(originalFileName).name}+${count}${parse(originalFileName).ext}`;
```
Comment (mertalev) [inline server/src/domain/download/download.service.ts:99]:
```suggestion
        const parsedFilename = parse(originalFileName);
        filename = `${parsedFilename.name}+${count}${parsedFilename.ext}`;
```

**Review by mertalev — APPROVED** [review]

**Comment by jrasm91** [comment]
Can you fix this?

https://github.com/immich-app/immich/blob/3da2b05428c120b52b35a697c7cabd42f6ce370b/web/src/lib/utils/asset-utils.ts#L162

**Pull request merged**